tests/ui/len_zero.stderr

```diff
@@ -1,11 +1,13 @@
 error: item `PubOne` has a public `len` method but no corresponding `is_empty` method
-  --> $DIR/len_zero.rs:10:5
+  --> $DIR/len_zero.rs:9:1
    |
-10 |       pub fn len(self: &Self) -> isize {
-   |  _____^ starting here...
+9  |   impl PubOne {
+   |  _^ starting here...
+10 | |     pub fn len(self: &Self) -> isize {
 11 | |         1
 12 | |     }
-   | |_____^ ...ending here
+13 | | }
+   | |_^ ...ending here
```

Comment:
I've pushed a change which:

* adds an extra, irrelevant impl block to PubOne to check that we this doesn't generate any errors
* adds a PubAllowed struct equivalent to PubOne but with an `allow` attribution on the relevant impl.

Are you happy to merge this as is, or do you want to wait for a fix to the impls-in-different-blocks issue?